pkg/api/v1beta3/types_swagger_doc.go

```diff
@@ -0,0 +1,1019 @@
+		"":                     "Similar to VolumeSource but meant for the administrator who creates PVs. Exactly one of its members must be set.",
+		"gcePersistentDisk":    "GCEPersistentDisk represents a GCE Disk resource that is attached to a kubelet's host machine and then exposed to the pod.",
+		"awsElasticBlockStore": "AWSElasticBlockStore represents an AWS Disk resource that is attached to a kubelet's host machine and then exposed to the pod.",
+		"hostPath":             "HostPath represents a directory on the host. This is useful for development and testing only. on-host storage is not supported in any way.",
+		"glusterfs":            "Glusterfs represents a Glusterfs volume that is attached to a host and exposed to the pod",
+		"nfs":                  "NFS represents an NFS mount on the host",
+		"rbd":                  "RBD represents a Rados Block Device mount on the host that shares a pod's lifetime",
+		"iscsi":                "ISCSI represents an ISCSI Disk resource that is attached to a kubelet's host machine and then exposed to the pod.",
+	}
+}
+
+func (PersistentVolumeSpec) SwaggerDoc() map[string]string {
+	return map[string]string{
+		"capacity":                      "Resources represents the actual resources of the volume",
+		"accessModes":                   "AccessModes contains all ways the volume can be mounted",
+		"claimRef":                      "ClaimRef is part of a bi-directional binding between PersistentVolume and PersistentVolumeClaim. ClaimRef is expected to be non-nil when bound. claim.VolumeName is the authoritative bind between PV and PVC.",
+		"persistentVolumeReclaimPolicy": "Optional: what happens to a persistent volume when released from its claim.",
+	}
+}
+
+func (PersistentVolumeStatus) SwaggerDoc() map[string]string {
+	return map[string]string{
+		"phase":   "Phase indicates if a volume is available, bound to a claim, or released by a claim",
+		"message": "A human-readable message indicating details about why the volume is in this state.",
+		"reason":  "Reason is a brief CamelCase string that describes any failure and is meant for machine parsing and tidy display in the CLI",
+	}
+}
+
+func (Pod) SwaggerDoc() map[string]string {
+	return map[string]string{
+		"":       "Pod is a collection of containers that can run on a host. This resource is created by clients and scheduled onto hosts.",
+		"spec":   "Spec defines the behavior of a pod.",
+		"status": "Status represents the current information about a pod. This data may not be up to date.",
+	}
+}
+
+func (PodCondition) SwaggerDoc() map[string]string {
+	return map[string]string{
+		"type":   "Type is the type of the condition",
+		"status": "Status is the status of the condition",
+	}
+}
+
+func (PodConditionType) SwaggerDoc() map[string]string {
+	return map[string]string{
+		"": "PodConditionType is a valid value for PodCondition.Type",
+	}
+}
+
+func (PodExecOptions) SwaggerDoc() map[string]string {
+	return map[string]string{
```

Comment:
I couldn't find a way to declare a static map. Go just doesn't support that as far as I can tell.

IMHO I wouldn't worry that much though as those functions are called only once on booting time of the api-server and for every object we construct only one of those maps. Creating documentation for each of the fields doesn't require a new doc-map. So in any case this is optimal as the structs are inspected only once from `go-restful` during booting time and generating documentation for the fields we are reusing the same map. Thus, `SwaggerDoc()` is called only once during lifetime of api-server.